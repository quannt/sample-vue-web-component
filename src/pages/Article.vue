<template>
  <div class="container" v-html="this.htmlContent">
  </div>
</template>

<script>
const url = "https://www.interaction-design.org/widgets/articles?ep=usabilitygeek";

export default {
  name: "Article",
  components: {

  },
  data() {
    return {
      htmlContent: ""
    }
  },
  mounted()  {
    window.fetch(url).then((response) => {
      return response.text()
    }).then((body) => {
      this.htmlContent = body;
    })
  }
}
</script>

<style lang="scss">
.articlesWidget {
  padding: 0 10px;
}
.articlesWidget__logo {
  display: flex;
  justify-content: center;
}

.articlesWidget__header {
  font-size: 16px;
  font-weight: bold;
  font-stretch: condensed;
  text-transform: uppercase;
  color: #404040;
  display: table-cell;
  border-bottom: 1px solid #009cde;

  &:after {
    content: "";
    display: table-cell;
    width: 100%;
  }
}

.articlesWidget__articles {
  display: flex;
  flex-direction: column;
  list-style: none;

  .article {
    display: flex;
    padding: 10px;
    border-top: 2px solid #dddddd;
    a {
      text-decoration: none;
      font-size: 16px;
      line-height: 1.25;
      color: #404040;
    }

    .article__fulltext {
      &:after {
        display: block;
        content: ' ';
        background-image: url('../assets/image/angle-right.svg');
        background-size: 22px 22px;
        height: 22px;
        width: 22px;
        flex-shrink: 0;
        transition: transform .25s cubic-bezier(.175,.885,.32,1.275);
      }

      display: flex;
      align-items: center;
    }

    .article__title {
      display: flex;
      flex-direction: column;

      .article__publishedTime {
        font-size: 10px;
        font-stretch: condensed;
        color: #848484;
      }
    }

    &:hover {
      background-color: #f9f9f9;


      .article__fulltext {
        &:after {
          transform: translateX(5px);
        }
      }
    }
    transition: background-color .21s cubic-bezier(.455,.03,.515,.955);

  }
}

.articlesWidget__loadMore {
  display: flex;
  justify-content: center;
  padding-top: 10px;

  button {
    border-radius: 5px;
    box-shadow: 0 1px 3px 0 rgba(0, 0, 0, 0.3);
    background-color: #dddddd;

    font-size: 14px;
    font-stretch: condensed;
    text-align: center;
    color: #404040;
    padding: 8px 23px;
  }
}

.articlesWidget__text {
  padding-top: 20px;
  padding-bottom: 20px;
  font-size: 16px;
  line-height: 1.6;
  color: #404040;

  a {
    text-decoration: none;
    color: #009cde;
  }
}
</style>
