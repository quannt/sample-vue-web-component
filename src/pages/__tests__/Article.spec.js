import { shallowMount } from "@vue/test-utils";
import Article from "../Article.vue";

const htmlContent = `<div class="articlesWidget">
    <div class="articlesWidget__logo">
        <img src="https://public-media.interaction-design.org/images/idf-logo-full.svg" alt="IDF logo" width="200px">
    </div>
    <h2 class="articlesWidget__header">Latest UX daily articles</h2>
    <ul class="articlesWidget__articles">
                    <li class="article">
    <a href="https://www.interaction-design.org/literature/article/don-t-make-me-think-key-learning-points-for-ux-design-for-the-web?ep=usabilitygeek&amp;utm_source=uxdailywidget&amp;utm_medium=widget&amp;utm_campaign=partners&amp;utm_content=usabilitygeek"
       target="_blank"
       class="article__link">
        <div class="article__fulltext">
            <div class="article__title">
                Don’t Make Me Think – Key Learning Points for UX Design for the Web
                <time datetime="1547564400" class="article__publishedTime">
                    1 hour ago
                </time>
            </div>
        </div>
    </a>
</li>
                    <li class="article">
    <a href="https://www.interaction-design.org/literature/article/the-basics-of-recruiting-users-for-usability-testing?ep=usabilitygeek&amp;utm_source=uxdailywidget&amp;utm_medium=widget&amp;utm_campaign=partners&amp;utm_content=usabilitygeek"
       target="_blank"
       class="article__link">
        <div class="article__fulltext">
            <div class="article__title">
                The Basics of Recruiting Users for Usability Testing
                <time datetime="1547553600" class="article__publishedTime">
                    4 hours ago
                </time>
            </div>
        </div>
    </a>
</li>
                    <li class="article">
    <a href="https://www.interaction-design.org/literature/article/7-ux-deliverables-what-will-i-be-making-as-a-ux-designer?ep=usabilitygeek&amp;utm_source=uxdailywidget&amp;utm_medium=widget&amp;utm_campaign=partners&amp;utm_content=usabilitygeek"
       target="_blank"
       class="article__link">
        <div class="article__fulltext">
            <div class="article__title">
                7 UX Deliverables: What will I be making as a UX designer?
                <time datetime="1547553600" class="article__publishedTime">
                    4 hours ago
                </time>
            </div>
        </div>
    </a>
</li>
                    <li class="article">
    <a href="https://www.interaction-design.org/literature/article/when-should-i-conduct-usability-testing-for-a-product?ep=usabilitygeek&amp;utm_source=uxdailywidget&amp;utm_medium=widget&amp;utm_campaign=partners&amp;utm_content=usabilitygeek"
       target="_blank"
       class="article__link">
        <div class="article__fulltext">
            <div class="article__title">
                When Should I Conduct Usability Testing for a Product?
                <time datetime="1547550000" class="article__publishedTime">
                    5 hours ago
                </time>
            </div>
        </div>
    </a>
</li>
            </ul>

    <div class="articlesWidget__loadMore">
        <button type="button">Load more articles</button>
    </div>

    <div class="articlesWidget__text">
        UX Daily is the world's largest free online resource on UX Design by the
        <a href="https://www.interaction-design.org/invite?ep=usabilitygeek&amp;utm_source=uxdailywidget&amp;utm_medium=widget&amp;utm_campaign=partners&amp;utm_content=usabilitygeek" target="_blank">Interaction Design Foundation</a>.

        Become a member and
        <a href="https://www.interaction-design.org/invite?ep=usabilitygeek&amp;utm_source=uxdailywidget&amp;utm_medium=widget&amp;utm_campaign=partners&amp;utm_content=usabilitygeek" target="_blank">
            get 3 months free</a>!
            </div>
</div>`;

beforeEach(function() {
  global.fetch = jest.fn().mockImplementation(() => {
    return new Promise(resolve => {
      resolve({});
    });
  });
});
describe("Article.vue", () => {
  test("render articles list", () => {
    const wrapper = shallowMount(Article);
    wrapper.setData({ htmlContent: htmlContent, isLoading: false });

    expect(wrapper.findAll(".article")).toHaveLength(4);
  });

  test("renders articles list snapshot correctly", () => {
    const wrapper = shallowMount(Article);
    wrapper.setData({ htmlContent: htmlContent, isLoading: false });

    expect(wrapper.element).toMatchSnapshot();
  });
});
