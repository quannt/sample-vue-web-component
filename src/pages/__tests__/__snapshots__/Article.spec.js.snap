// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Article.vue renders articles list snapshot correctly 1`] = `
<div>
  <content-loader-stub
    animate="true"
    height="160"
    preserveaspectratio="xMidYMid meet"
    primarycolor="#f3f3f3"
    secondarycolor="#ecebeb"
    speed="2"
    width="400"
  >
    <rect
      height="46"
      rx="0"
      ry="0"
      width="230.46"
      x="88.5"
      y="17.2"
    />
     
    <rect
      height="7"
      rx="0"
      ry="0"
      width="323"
      x="40.5"
      y="83.2"
    />
     
    <rect
      height="7"
      rx="0"
      ry="0"
      width="323"
      x="40.5"
      y="104.2"
    />
     
    <rect
      height="7"
      rx="0"
      ry="0"
      width="323"
      x="40.5"
      y="124.2"
    />
     
    <rect
      height="7"
      rx="0"
      ry="0"
      width="323"
      x="39.5"
      y="141.2"
    />
  </content-loader-stub>
   
  <div
    class="container"
  >
    <div
      class="articlesWidget"
    >
      
    
      <div
        class="articlesWidget__logo"
      >
        
        
        <img
          alt="IDF logo"
          src="https://public-media.interaction-design.org/images/idf-logo-full.svg"
          width="200px"
        />
        
    
      </div>
      
    
      <h2
        class="articlesWidget__header"
      >
        Latest UX daily articles
      </h2>
      
    
      <ul
        class="articlesWidget__articles"
      >
        
                    
        <li
          class="article"
        >
          
    
          <a
            class="article__link"
            href="https://www.interaction-design.org/literature/article/don-t-make-me-think-key-learning-points-for-ux-design-for-the-web?ep=usabilitygeek&utm_source=uxdailywidget&utm_medium=widget&utm_campaign=partners&utm_content=usabilitygeek"
            target="_blank"
          >
            
        
            <div
              class="article__fulltext"
            >
              
            
              <div
                class="article__title"
              >
                
                Don’t Make Me Think – Key Learning Points for UX Design for the Web
                
                <time
                  class="article__publishedTime"
                  datetime="1547564400"
                >
                  
                    1 hour ago
                
                </time>
                
            
              </div>
              
        
            </div>
            
    
          </a>
          

        </li>
        
                    
        <li
          class="article"
        >
          
    
          <a
            class="article__link"
            href="https://www.interaction-design.org/literature/article/the-basics-of-recruiting-users-for-usability-testing?ep=usabilitygeek&utm_source=uxdailywidget&utm_medium=widget&utm_campaign=partners&utm_content=usabilitygeek"
            target="_blank"
          >
            
        
            <div
              class="article__fulltext"
            >
              
            
              <div
                class="article__title"
              >
                
                The Basics of Recruiting Users for Usability Testing
                
                <time
                  class="article__publishedTime"
                  datetime="1547553600"
                >
                  
                    4 hours ago
                
                </time>
                
            
              </div>
              
        
            </div>
            
    
          </a>
          

        </li>
        
                    
        <li
          class="article"
        >
          
    
          <a
            class="article__link"
            href="https://www.interaction-design.org/literature/article/7-ux-deliverables-what-will-i-be-making-as-a-ux-designer?ep=usabilitygeek&utm_source=uxdailywidget&utm_medium=widget&utm_campaign=partners&utm_content=usabilitygeek"
            target="_blank"
          >
            
        
            <div
              class="article__fulltext"
            >
              
            
              <div
                class="article__title"
              >
                
                7 UX Deliverables: What will I be making as a UX designer?
                
                <time
                  class="article__publishedTime"
                  datetime="1547553600"
                >
                  
                    4 hours ago
                
                </time>
                
            
              </div>
              
        
            </div>
            
    
          </a>
          

        </li>
        
                    
        <li
          class="article"
        >
          
    
          <a
            class="article__link"
            href="https://www.interaction-design.org/literature/article/when-should-i-conduct-usability-testing-for-a-product?ep=usabilitygeek&utm_source=uxdailywidget&utm_medium=widget&utm_campaign=partners&utm_content=usabilitygeek"
            target="_blank"
          >
            
        
            <div
              class="article__fulltext"
            >
              
            
              <div
                class="article__title"
              >
                
                When Should I Conduct Usability Testing for a Product?
                
                <time
                  class="article__publishedTime"
                  datetime="1547550000"
                >
                  
                    5 hours ago
                
                </time>
                
            
              </div>
              
        
            </div>
            
    
          </a>
          

        </li>
        
            
      </ul>
      

    
      <div
        class="articlesWidget__loadMore"
      >
        
        
        <button
          type="button"
        >
          Load more articles
        </button>
        
    
      </div>
      

    
      <div
        class="articlesWidget__text"
      >
        
        UX Daily is the world's largest free online resource on UX Design by the
        
        <a
          href="https://www.interaction-design.org/invite?ep=usabilitygeek&utm_source=uxdailywidget&utm_medium=widget&utm_campaign=partners&utm_content=usabilitygeek"
          target="_blank"
        >
          Interaction Design Foundation
        </a>
        .

        Become a member and
        
        <a
          href="https://www.interaction-design.org/invite?ep=usabilitygeek&utm_source=uxdailywidget&utm_medium=widget&utm_campaign=partners&utm_content=usabilitygeek"
          target="_blank"
        >
          
            get 3 months free
        </a>
        !
            
      </div>
      

    </div>
  </div>
</div>
`;
